### DISCUSSION TITLE
Color matrix entries

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#4202] [Q&A] [answered]
Consider the following code:

```
$
mat(
  1, 2, 3;
  4, 5, 6;
  7, 8, 9;
)
$
```

Is it possible to color the background of entries 1, 2, 4, 5 ?

**Comment by TS60** [comment, marked as answer]
It's not possible in math mode and not possible without hacks in text mode, I think.

```typst
#math.mat(
  table(
    stroke: none,
    columns: (10pt, 10pt, 10pt),
    inset: 3pt,
    align: center,
    fill: (x, y) =>
    if x <= 1 and y <= 1 { luma(240) }
    else { none },
    [1],
    [2],
    [3],
    [4],
    [5],
    [6],
    [7],
    [8],
    [9],
  )
)
```
![colored_cell_matrix](https://github.com/typst/typst/assets/152509161/eaacb2f0-dbec-4f63-92f1-4ecf6574a833)

**Comment by psads-git** [comment]
Nice trick, @TS60!  Thanks!
